lib/graphql/static_validation/rules/argument_literals_are_compatible.rb

```diff
@@ -6,22 +6,52 @@ def validate_node(parent, node, defn, context)
         return if node.value.is_a?(GraphQL::Language::Nodes::VariableIdentifier)
         arg_defn = defn.arguments[node.name]
         return unless arg_defn
+        if !context.schema.error_bubbling
```

Comment:
I see the branches below have the same _happy path_, but different error handling, is that right? It looks like `LiteralValidationError` is only raised when `context.schema.error_bubbling` (because of the check in `maybe_raise_if_invalid`, is that right? 

If those assumptions are true, could this `if ... else ... end` be refactored to 

```
begin 
  # happy path as before
rescue GraphQL::CoercionError => err 
  # as before 
rescue GraphQL::LiteralValidationError => 
  # new error handling code 
end 
```

? If so, that would really help me keep track of what different behaviors we expect in different cases, and reduce some duplication!

Reply:
i think i tried this at one point and ended up introducing some other bugs but it certainly seems like it should work. i'll try again now.

Reply:
ok - cleaned it up and found/fixed a bug that was somehow getting handled properly before.